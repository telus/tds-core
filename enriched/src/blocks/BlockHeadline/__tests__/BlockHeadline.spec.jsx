import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Headline from '../';

describe('Headline', () => {
  const data = { eyebrow: 'Foo', title: 'Bar' };

  it('matches snapshot', () => {
    const wrapper = mount(<Headline {...data} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should render eyebrow', () => {
    const wrapper = mount(<Headline {...data} />);
    expect(wrapper.find('p').first().render().text()).toEqual('Foo');
  });

  it('should render without subcategory', () => {
    const wrapper = mount(<Headline title={data.title} />);
    expect(wrapper.find('p').first().render().text()).toEqual('');
  });

  it('should render title', () => {
    const wrapper = mount(<Headline {...data} />);
    expect(wrapper.find('h1').first().render().text()).toEqual('Bar');
  });

  it('should use headline style', () => {
    const wrapper = mount(<Headline {...data} />);
    expect(wrapper.find('section').first().hasClass('b-headline')).toBeTruthy();
  });

  it('should render subtext', () => {
    const wrapper = mount(<Headline {...data} subtext="These are the Foo Bars in your area." />);
    expect(wrapper.find('p').last().render().text()).toEqual('These are the Foo Bars in your area.');
  });

  it('should only render subtext p tag if present', () => {
    const wrapper = mount(<Headline {...data} />);
    expect(wrapper.find('p').length).toEqual(1);
  });

  it('should remove bottom margin of title when subtext is not present', () => {
    const wrapper = mount(<Headline {...data} />);
    expect(wrapper.find('h1').first().hasClass('bottom-margin-0')).toEqual(true);
  });

  it('should remove bottom margin of subtext when present but not title', () => {
    const wrapper = mount(<Headline {...data} subtext="These are the Foo Bars in your area." />);
    expect(wrapper.find('p').last().hasClass('bottom-margin-0')).toEqual(true);
    expect(wrapper.find('h1').last().hasClass('bottom-margin-0')).toEqual(false);
  });

});
