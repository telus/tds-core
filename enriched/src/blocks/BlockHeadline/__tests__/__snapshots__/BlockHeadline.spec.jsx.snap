// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline matches snapshot 1`] = `
<BlockHeadline
  eyebrow="Foo"
  subtext=""
  title="Bar"
>
  <section
    className="b-headline"
  >
    <div
      className="container container--limited-width"
    >
      <div
        className="grid-row"
      >
        <div
          className="medium-6 large-7 xl-7"
        >
          <p
            className="heading-4"
          >
            Foo
          </p>
          <h1
            className="display-heading-1 bottom-margin-0"
          >
            Bar
          </h1>
        </div>
      </div>
    </div>
  </section>
</BlockHeadline>
`;
